ctp2_data/default/gamedata/Units.txt

```diff
@@ -879,7 +879,7 @@ UNIT_CLERIC {
    EnableAdvance ADVANCE_THEOLOGY
    ObsoleteAdvance ADVANCE_MASS_MEDIA
    UpgradeTo UNIT_TELEVANGELIST
-   GovernmentType GOVERNMENT_THEOCRACY
+#   GovernmentType GOVERNMENT_THEOCRACY

```

Comment:
This change seems to be introduced with 6a57d7a5208ee32c8337bace478a217094448c7d but does not originate from either parent of the merge. Unless unavoidable, this is very bad! There seem to be other changes so I created a plain merge (https://github.com/LynxAbraxas/civctp2/commit/6c51019793152e4c18b1dfdf9c2495dc6611ea6a) and a commit with the extra changes (https://github.com/LynxAbraxas/civctp2/commit/523f586adda0e96a0e5c6ed213c6b835e9a93b0e). Since even these do not seem related I'd ask you to make the effort of splitting the changes into their own meaningful commits. Otherwise the changes are hardly trackable nor understandable. For example I'd make the changes concerning Clerics and Televangelists in all govs a commit on its own, even if it is just 2 lines commented. This makes a big change and it is very helpful for future development if this is in its own commit with a good commit message, that contains a link to the discussion we had on this on GH, e.g. #129. 
Though I am trying hard to do good commit messages, I know they are not always as meaningful to others, for which @DEVoytas  pointed me to (https://github.com/civctp2/civctp2/pull/14#commitcomment-31035813):
https://chris.beams.io/posts/git-commit/
https://github.com/erlang/otp/wiki/writing-good-commit-messages :)

Reply:
Thanks for pointing that out and for the clean merge commit.

Indeed 6a57d7a is quite messy, I have to see whether TortoiseGit, has a clear interface for it. And whether the mistake comes from there. Basically, the interface should show me somewhere the changes that do not come from the parents. So that means I have to clean up, here.

Basically, these changes are supposed for debugging or experimental. The extra changes of [LynxAbraxas@523f586] are not supposed to be in. So far I was aware of ctpai.cpp and civ3_main.cpp, these changes for debugging, and if it were those alone, I could still think about to leave them alone, but the other stuff is not supposed to be in.

Actually, units.txt is not even a commit of its own but also a pull request, since it changes the game play.

directoractions.h is just something outcommented, I had in the working copy. It is supposed to fix a memory leak, but it makes the game crash.

Then we have strategies.txt, that are 2 or 3 commits, one of them goes into its own PR, since I still don't know whether that is the best to handle it, and it also has a big effect.

In govern.txt, I increased the city limit to see whether the AI can conquer the whole world, that still doesn't work. It still has some trouble with underwater cities. So that is something for debugging, but not for this PR.

Same with units.txt, whether the AI can use Clerics properly, is within the scope of this PR, but not whether it can build it.

Reply:
> Thanks for pointing that out and for the clean merge commit.

Just noticed that I also screwed up the merge commit message:
```
Fixed conflicts in:
    ../ctp2_code/gs/gameobj/CityData.cpp
    with
    git checkout 6a57d7a5208ee32c8337bace478a217094448c7d -- ctp2_code/gs/gameobj/CityData.cpp
```
Should be in the message for https://github.com/LynxAbraxas/civctp2/commit/6c51019793152e4c18b1dfdf9c2495dc6611ea6a

> I have to see whether TortoiseGit, has a clear interface for it. And whether the mistake comes from there. Basically, the interface should show me somewhere the changes that do not come from the parents.

Never used TortoiseGit so far, so can't help you there, though simple `gitk` from the git-package isn't very fancy, it does show this quite nicely:

6a57d7a:
![ss_2019-08-30_14:24:51](https://user-images.githubusercontent.com/21012234/64020855-ef978a00-cb32-11e9-9eb0-dfdfdb56c1d8.png)

in contrast for https://github.com/LynxAbraxas/civctp2/commit/6c51019793152e4c18b1dfdf9c2495dc6611ea6a `CityData.cpp` is the only one with changes from both parents, i.e. had a merge conflict:
![ss_2019-08-30_14:26:02](https://user-images.githubusercontent.com/21012234/64020903-09d16800-cb33-11e9-8da7-98f914c737bd.png)

> Actually, units.txt is not even a commit of its own but also a pull request, since it changes the game play.

Yes, I would welcome a PR on its own here, too.

> Same with units.txt, whether the AI can use Clerics properly, is within the scope of this PR, but not whether it can build it.

Not sure if I really get here what you mean. I'll test this PR with this to see if conversions by the AI happen or if there are any bugs like there were for franchise and sue franchise.  Before this PR is merged I'd say the changes to units.txt should be reverted (which is just a `git revert SHA` if the change were in its own commit), and a new PR just concerning clerics/televangelists should be used to discuss how to best realize this.

Reply:
> Just noticed that I also screwed up the merge commit message:
> 
> ```
> Fixed conflicts in:
>     ../ctp2_code/gs/gameobj/CityData.cpp
>     with
>     git checkout 6a57d7a5208ee32c8337bace478a217094448c7d -- ctp2_code/gs/gameobj/CityData.cpp
> ```

That should have gone into the merge commit, at least the message says merge commit. Since I have to tear apart this commit, I won't use that commit and message anyway.

Reply:
Indeed TortoiseGit is not so nice a bout merge commits, it shows me the files by parent, but actually the changes that come from no parent are shown as they come from both parents. Even so they are supposed to be in an extra section.

And also when I committed the merge commit, I could decide what file goes in, but I supposed that it would not put in the files that come from no parent.